Available zones discovery for azure (#3337)
kyma-project/kyma-environment-broker: docs/contributor/03-55-zones-discovery.md

The Zones Discovery feature extends Kyma Environment Broker (KEB) to dynamically determine availability zones for both the Kyma worker node pool and additional worker node pools during provisioning and updates.
Operators can configure worker node pools to use either static zone assignments (predefined in the configuration) or dynamic zone assignments (queried live from the hyperscaler).

> This feature is supported on **AWS** and **Azure**.

Reviewer: ```suggestion
> ### Note:
> This feature is only supported on Amazon Web Services and Microsoft Azure.
```

Reviewer: Let's stick to the Note format for better visibility.

```json lines
{"level":"WARN", "msg":"Provider aws has zones discovery enabled, but region us-west-2 is configured with 4 static zone(s), which will be ignored."} 
{"level":"WARN", "msg":"Provider azure has zones discovery enabled, but region westeurope is configured with 3 static zone(s), which will be ignored."}
```

## Credentials

> KEB resolves hyperscaler credentials from Gardener secrets referenced by a `CredentialsBinding`. The secret fields follow the Gardener convention for each provider type.

Reviewer: ```suggestion
KEB resolves hyperscaler credentials from Gardener Secrets referenced by a `CredentialsBinding`. The Secret fields follow the Gardener convention for each provider type.
```

Reviewer: How about rephrasing to "The fields within a Secret" for more clarity? If that's what you mean.

Author: sure

To optimize performance, if the same machine type is used in multiple worker node pools, KEB queries the hyperscaler only once per unique machine type and reuses the result across all occurrences. This solution eliminates unnecessary duplicate calls.
The subscription secret is used only for validation. Its name is logged to support traceability in case of validation failures.

> ### Azure — zone restrictions

Reviewer: ```suggestion
### Microsoft Azure — Zone Restrictions
```
